Totals use fewer tables when possible

LindsayYoung opened the pull request:
This make the call a bit faster by automatically reducing the amount of tables if the call does not need them all. I also reduced the amount of fields that show up automatically.

cmc333333 commented on webservices/db.py:
```diff
@@ -26,3 +26,18 @@ def htsql_conn():
         g._htsql_conn = HTSQL(htsql_conn_string)
     return g._htsql_conn
+
+def as_dicts(data):
```
If you're moving this here (seems like it makes more sense), remember to remove the version in `webservices/resources.py`

LindsayYoung replied:
Good catch, I am doing that now!

cmc333333 commented on webservices/totals/models.py:
```diff
@@ -48,13 +40,48 @@ def query_text(self, show_fields):
+            for d in str(data):
+                if d.isalpha() == True:
+                    if d not in types:
+                        types.append(d)
```
Don't know if you'd like this one-liner better:

```python
types = set(d for d in str(data) if d.isalpha())
```

LindsayYoung replied:
Good point. I will do that.

cmc333333 commented:
I don't understand the underlying data well enough to validate, but it looks like this divides what type of query it makes based on the request, which no doubt removes a lot of unneeded work. :+1:

LindsayYoung commented:
@cmc333333 thanks for the suggestions. They should all be incorporated now.

I also fixed #337 while I was in there and added a test. The API will return a sane format when no search results are found.

LindsayYoung commented:
just merged this in to the older pr about totals
